roofnet/README.md

```diff
@@ -0,0 +1,63 @@
+# Summary
+Roofnet is a wireless mesh architecture that provides high performance Internet access while demanding little deployment planning or operational management.
+
+## Goals
+1. Volunteer users host nodes at home. Thus, no central control over topology, and no central planning.
+2. Omnidirectional antennas that are low quality and probably have high interference.
+3. Multi-hop routing.
+4. High TCP throughout 
+
+## Node Addresses
+Wireless interface IP addresses are auto configured, using a private prefix (e.g. 10) for the high byte, and the lower 24 bits of ethernet addresses for the lower three bytes.
+
+A roofnet node uses NAT between wired ethernet and Roofnet, so that connections from a user's host appear to the rest of Roofnet to have originated from the user's node. This means that no address allocation coordination across Roofnet boxes is required.
+
+## Internet Gateways
+Roofnet assumes that a small fraction of Roofnet users will voluntarily share their wired Internet acces links.
+
+A roofnet node sends a DHCP request on ethernet to test if it is an internet gateway. If this succeeds, the node advertises itself as a gateway. Other nodes will use this to send traffic to the internet.
+
+Roofnet nodes open a TCP connection to the route with the best metric, and track the gateway used for each open TCP connection they originate.
+
+## Hop Count
+Practical studies found that minimum-hop-count routes are significantly throughput-suboptimal, so we must choose paths with care. This is hard to predict because link loss is often asymmetric, and there is a wide range of loss rates.
```

Comment:
ie just say

Practical studies found that minimum-hop-count routes are significantly throughput-suboptimal: link loss can be high-variance and asymmetric.